src/components/allNews/AllNewHeader.js

```diff
@@ -1,45 +1,47 @@
+import Component from '../core/Component.js';
 import Icon from '../common/Icon.js';
-
-export default class AllNewHeader {
-  constructor() {
-    this.$header = document.createElement('div');
-    this.$header.className = 'all-news-header';
-
-    this.render();
-
-    return this.$header;
+import { customQuerySelector } from '../../utils/index.js';
+import { toggleDarkMode } from '../../index.js';
+
+export default class AllNewHeader extends Component {
+  setup() {
+    const isCurrentDarkMode = document.body.className === 'dark';
+    this.state = {
+      modeIcon: isCurrentDarkMode ? 'moon' : 'sun',
+    };
   }
-
-  render() {
-    this.addTitleNavigator();
-    this.addIconNavigator();
+  template() {
+    return `<nav class='view-type-wrapper'>
+            <span class='selected-bold16 text-strong'>전체 언론사</span>        
+            <span class='available-medium16 text-weak'>내가 구독한 언론사</span>
+            </nav>
+
+            <div class='view-type-icon'>
+              <img id="darkmode-icon" src="src/assets/icons/${this.state.modeIcon}.png" />
+              <img id ='list-view-icon' class='icon-medium'/>
+              <img id ='grid-view-icon' class='icon-medium'/>
+            </div>`;
   }
 
-  addTitleNavigator() {
-    const $titleNavigation = document.createElement('nav');
-    $titleNavigation.className = 'view-type-wrapper';
-
-    const $allPress = document.createElement('span');
-    const $subscibedPress = document.createElement('span');
+  mounted() {
+    const listIconName = this.props.view === 'list' ? 'list-view-focus' : 'list-view';
```

Comment:
grid나 list같이 자주 사용되는 문저열 상수로 관리해도 좋을것 같습니다~

```suggestion
const mode = {
  GRID:'grid',
  LIST:'list'
}
```

Reply:
반복되는 상수는 위처럼 구현하는게 오타를 줄일 수 있는 방법일 거 같네요 ㅎㅎ 반영하겠습니다!!!